### ISSUE TITLE
Hunter + Warlock pets positioning issue when attacking target. 

### ISSUE THREAD

**Issue by dogmax** [LightsHope/issues#799]
**Description of issue:** When I send my pet to attack attack a hostile mob, it will sometimes continue to walk past the mob while the mob stays in the position where they were supposed to engage in a melee fight.

**Steps to reproduce:** I am unable to reproduce the bug reliably, but here is a description of the last occurence of the bug

1) Player is a lvl 16 gnome warlock in Loch Modan. 

2) Walking towards a lvl 11 spider at coordinates 34,54

3) Player sends in Voidwalker pet to engage lvl 11 spider _while_ player continuously walks towards the spider. 

4) Player stops walking before Voidwalker hits lvl 11 spider

5) Voidwalker continues into the distance while lvl 11 spider stays put and does melee dmg to the Voidwalker. 

**Personal thoughts:** What I think influences this behaviour based on unsuccessful attempts at reproducing the bug at will: 

a) Hitbox mismatch between mob and pet

b) Player movement while pet is engaging the mob. When does the player stop moving? Who aggroed first? Stuff like that.

c) Might be related to the Death Coil bug, where an enemy player exhibits similar behaviour in PvP combat under the effect of Death Coil.

Please fix.

**Comment by derekvcsr** [comment]
Sadly, this was a big issue with all pets in early Vanilla. Hunters could see the same issue. I don't recall when they semi-fixed in vanilla.

**Comment by Cabro** [comment]
No, this is a bug that I noticed and reported several times only to be bypassed on both previous bug trackers. It has something to do with how units handle repositioning. All NPCs on the server are affected by this bug. They just don't stop when they reach their target, they continue their initial path until they are out of their target's melee range. The really buggy part is that it checks for outside of the melee range even if you are moving while it is moving, and in that case it will only reposition when its original path (patrol has a turn at some point on the road and they reached it) would change, or you let it go out of your melee range. This is why mobs after taunt keep moving to their initial target, while targetting and attacking the new target that has taunted them.
Pets are also handled with this mechanic, and suffer from the same bug. Hence the behavior you noticed.
It's a whole mess, which i doubt will ever get fixed.
